base/dict.jl

```diff
@@ -189,6 +189,12 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict)
+
+Create an empty dictionary with the same key and value types as `d`.
+"""
 similar{K,V}(d::Dict{K,V}) = Dict{K,V}()
 similar{K,V}(d::Dict, ::Type{Pair{K,V}}) = Dict{K,V}()
```

Comment:
@pabloferz You have answered the question I should have asked!

My confusion is why I am not able to write `e=Dict{eltype(d)}()` to create a dictionary such that `eltype(d) == eltype(e)`, in the same way as `T=eltype{vector{T}()}`.

This is nitpicking but shouldn't the signatures be something like
```
 similar{K,V}(::Dict{K,V}) = Dict{K,V}()
 similar{K,V}(::Dict, ::Type{Pair{K,V}}) = Dict{K,V}()
```
since any information about `d` except that it is a `Dict` is irrelevant?